src/stacking.rs

```diff
@@ -0,0 +1,52 @@
+
+use imp_prelude::*;
+use error::{ShapeError, ErrorKind, from_kind};
+
+/// Stack arrays along the given axis.
+///
+/// *Panics* if axis is out of bounds.
```

Comment:
I think we can check the axis element so that it doesn't panic. We want the `Result` functions to be panic free I think. Isn't returning `ErrorKind::OutOfBounds` ok?)